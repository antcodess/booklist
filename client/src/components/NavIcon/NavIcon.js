import React from 'react';

import './NavIcon.scss';

export const NavIcon = ({ toggleNavHidden }) => {
  return (
    <div className='nav-icon' onClick={() => toggleNavHidden()}></div>
  );
};

export default NavIcon; 