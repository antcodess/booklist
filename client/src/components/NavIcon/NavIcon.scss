.nav-icon {
  background: url('../../assets/mobile-nav-icon.png');
  background-repeat: no-repeat;
  background-size: 40px;
  width: 40px;
  height: 100%;
  position: absolute;
  top: 8px;
  left: 207px;
  cursor: pointer;
  display: none;
}

@media only screen and (max-width: 720px) {
  .nav-icon {
    display: flex;
  } 
}

@media only screen and (max-width: 500px) {
  .nav-icon {
    /* top: 2px; */
    left: 206px;
    /* left: 230px; */
  }
}

@media only screen and (max-width: 480px) {
  .nav-icon {
    left: 207px;
  }
}

@media only screen and (max-width: 400px) {
  .nav-icon {
    background-size: 30px;
    top: 11px;
    left: 165px;
  }
}

@media only screen and (max-width: 260px) {
  .cart-icon {
    display: none;
  }

  .nav-icon {
    top: 2px;
  }
}