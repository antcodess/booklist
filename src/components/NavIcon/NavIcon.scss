.nav-icon {
  background: url('../../assets/mobile-nav-icon.png');
  background-repeat: no-repeat;
  background-size: 40px;
  width: 100%;
  height: 100%;
  position: absolute;
  top: 8px;
  left: 207px;
  cursor: pointer;
  display: none;
}
