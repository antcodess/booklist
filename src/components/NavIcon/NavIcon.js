import React from 'react';

import './NavIcon.scss';

const NavIcon = ({ navHidden, setNavHidden }) => {
  return (
    <div className='nav-icon' onClick={() => setNavHidden(!navHidden)}></div>
  );
};

export default NavIcon; 