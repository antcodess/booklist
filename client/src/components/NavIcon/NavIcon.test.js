import React from 'react';
import { shallow } from 'enzyme';
import { NavIcon } from './NavIcon';

describe('NavIcon Component', () => {
  let wrapper;
  let mockToggleNavHidden;
  let dummyFunction = jest.fn();
  
  beforeEach(() => {

    mockToggleNavHidden = jest.fn();

    const mockProps = {
      toggleNavHidden: mockToggleNavHidden
    };

    wrapper = shallow(<NavIcon { ...mockProps } />);
  });

  it('should render NavDropDown Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleCartHidden when clicked for nav-icon div', () => {
    wrapper.find('.nav-icon').simulate('click');
    expect(mockToggleNavHidden).toHaveBeenCalled();
  });

});
