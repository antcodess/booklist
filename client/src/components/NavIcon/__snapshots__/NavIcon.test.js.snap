// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavIcon Component should render NavDropDown Component 1`] = `ShallowWrapper {}`;
